include/deal.II/matrix_free/tools.h

```diff
@@ -787,6 +1052,42 @@ namespace MatrixFreeTools
             c_pools[v].row_lid_to_gid.size() *
               (n_fe_components == 1 ? n_components : 1),
             Number(0.0));
+
+        // check if fast path can be taken via FEEvaluation
+        bool use_fast_path = true;
+
+        for (unsigned int v = 0; v < n_lanes_filled; ++v)
+          {
+            auto &c_pool = c_pools[v];
+
+            for (unsigned int i = 0; i < c_pool.row.size() - 1; ++i)
+              {
+                if ((c_pool.row[i + 1] - c_pool.row[i]) > 1)
+                  {
+                    use_fast_path = false;
+                    break;
+                  }
+                else if (((c_pool.row[i + 1] - c_pool.row[i]) == 1) &&
+                         (c_pool.val[c_pool.row[i]] != 1.0))
+                  {
+                    use_fast_path = false;
+                    break;
+                  }
+              }
+
+            if (use_fast_path == false)
+              break;
+          }
+
+        if (use_fast_path)
+          {
+            temp_values.resize(phi->dofs_per_cell);
+            return;
+          }
+        else
+          {
+            temp_values.resize(0);
```

Comment:
I personally like 
```suggestion
            temp_values.clear();
```
more because it is more descriptive, but this is not a hard concern.